src/components/ChangePlan/Checkout/BillingInformation/BillingInformation.js

```diff
@@ -0,0 +1,299 @@
+    const [formSubmitted, setFormSubmitted] = useState(false);
+    const [states, setStates] = useState([]);
+    const [sameAddressInformation, setSameAddressInformation] = useState(false);
+    const createTimeout = useTimeout();
+    const intl = useIntl();
+
+    useEffect(() => {
+      const fetchData = async () => {
+        const contactInformationResult = await dopplerUserApiClient.getContactInformationData();
+        const billingInformationResult =
+          await dopplerBillingUserApiClient.getBillingInformationData();
+        const statesResult = await staticDataClient.getStatesData(
+          billingInformationResult.success
+            ? billingInformationResult.value.country ?? contactInformationResult.value.country
+            : 'ar',
+          intl.locale,
+        );
+
+        setStates(statesResult.success ? statesResult.value : []);
+
+        const billingInformation = billingInformationResult.success
+          ? billingInformationResult.value
+          : { sameAddressAsContact: true };
+        if (billingInformation?.sameAddressAsContact) {
+          billingInformation.firstname = contactInformationResult.value.firstname;
+          billingInformation.lastname = contactInformationResult.value.lastname;
+          billingInformation.address = contactInformationResult.value.address;
+          billingInformation.city = contactInformationResult.value.city;
+          billingInformation.province = contactInformationResult.value.province;
+          billingInformation.country = contactInformationResult.value.country;
+          billingInformation.zipCode = contactInformationResult.value.zipCode;
+          billingInformation.phone = contactInformationResult.value.phone;
+        }
+
+        setState({
+          billingInformation,
+          success: billingInformationResult.success,
+          loading: false,
+          contactInformation: contactInformationResult.value,
+        });
+
+        setSameAddressInformation(billingInformation && billingInformation.sameAddressAsContact);
+      };
+      fetchData();
+    }, [dopplerUserApiClient, dopplerBillingUserApiClient, staticDataClient, intl.locale]);
+
+    const _ = (id, values) => intl.formatMessage({ id: id }, values);
+    const defaultOption = { key: '', value: _('checkoutProcessForm.empty_option_select') };
+    const language = intl.locale;
+
+    const _getFormInitialValues = () => {
+      let initialValues = getFormInitialValues(fieldNames);
+
+      if (state.billingInformation) {
+        initialValues = { ...state.billingInformation };
+      }
+
+      return initialValues;
+    };
+
+    const handleSameAddressChange = async (e, setFieldValue) => {
+      const sameAddress = !sameAddressInformation;
+      setSameAddressInformation(sameAddress);
+
+      if (sameAddress) {
+        if (state.contactInformation) {
+          setFieldValue(fieldNames.firstname, state.contactInformation.firstname);
```

Comment:
What do you think about something like this? do you think it could be better? or makes reading difficult?

```
let _fieldNames = { ...fieldNames };
delete _fieldNames.sameAddressAsContact;
Object.keys(fieldNames).forEach((key) => setFieldValue(fieldNames[key], state.contactInformation[key]))
```

Reply:
For me is a good approach